cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/detector/AnomalyDetectorState.java

```diff
@@ -0,0 +1,138 @@
+/*
+ * Copyright 2018 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.cruisecontrol.detector;
+
+import com.linkedin.cruisecontrol.detector.Anomaly;
+import com.linkedin.kafka.cruisecontrol.detector.notifier.AnomalyType;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+
+public class AnomalyDetectorState {
+  // Maximum number of anomalies to keep in the anomaly detector state.
+  private final static int NUM_RECENT_ANOMALIES = 10;
+  private static final String START_MS = "startMs";
+  private static final String VIOLATED_GOALS = "violatedGoals";
+  private static final String FAILED_BROKERS = "failedBrokers";
+  private static final String DESCRIPTION = "description";
+  private static final String SELF_HEALING_ENABLED = "selfHealingEnabled";
+  private static final String SELF_HEALING_DISABLED = "selfHealingDisabled";
+  private static final String RECENT_GOAL_VIOLATIONS = "recentGoalViolations";
+  private static final String RECENT_BROKER_FAILURES = "recentBrokerFailures";
+  private static final String RECENT_METRIC_ANOMALIES = "recentMetricAnomalies";
+  // Recent anomalies with record time by the anomaly type.
+  private final Map<AnomalyType, LinkedHashMap<Long, Anomaly>> _recentAnomaliesByType;
+  private final Map<AnomalyType, Boolean> _selfHealingEnabled;
+
+  public AnomalyDetectorState(Map<AnomalyType, Boolean> selfHealingEnabled) {
+    _recentAnomaliesByType = new HashMap<>(AnomalyType.cachedValues().size());
+    for (AnomalyType anomalyType : AnomalyType.cachedValues()) {
+      _recentAnomaliesByType.put(anomalyType, new LinkedHashMap<Long, Anomaly>() {
+        @Override
+        protected boolean removeEldestEntry(Map.Entry<Long, Anomaly> eldest) {
+          return this.size() > NUM_RECENT_ANOMALIES;
+        }
+      });
+    }
+    _selfHealingEnabled = selfHealingEnabled;
+  }
+
+  public void addAnomaly(AnomalyType anomalyType, Anomaly anomaly) {
+    _recentAnomaliesByType.get(anomalyType).put(System.currentTimeMillis(), anomaly);
+  }
+
+  public Map<AnomalyType, Boolean> selfHealingEnabled() {
+    return Collections.unmodifiableMap(_selfHealingEnabled);
+  }
+
+  public void setSelfHealingFor(AnomalyType anomalyType, boolean isSelfHealingEnabled) {
```

Comment:
I feel this method is unnecessary. For one thing I do not think we should change self healing setting from the state returned; for the other thing, change here will not change the config inside AnomalyNotifier( we are passed in `Collections.unmodifiableMap(_selfHealingEnabled)` when initiate this field here).